src/board.rs

```diff
@@ -115,6 +115,39 @@ impl Chessboard {
         self.white_turn = true;
     }
 
+    pub fn one_side_pieces(cb: &Chessboard, white: bool) -> u64 {
+        if white {
+            cb.white_bishops
+                & cb.white_king
+                & cb.white_knights
+                & cb.white_pawns
+                & cb.black_rooks
+                & cb.white_queen
+        } else {
+            cb.black_bishops
+                & cb.black_king
+                & cb.black_knights
+                & cb.black_pawns
+                & cb.black_rooks
+                & cb.black_queen
+        }
+    }
+
+    pub fn both_side_pieces(cb: &Chessboard) -> u64 {
+        cb.white_bishops
+            & cb.white_king
+            & cb.white_knights
+            & cb.white_pawns
+            & cb.black_rooks
+            & cb.white_queen
+            & cb.black_bishops
+            & cb.black_king
+            & cb.black_knights
+            & cb.black_pawns
+            & cb.black_rooks
+            & cb.black_queen
+    }
+
```

Comment:
simplify this to be `one_side_pieces(cb, true) & one_side_pieces(cb, false)`

Reply:
Done